touchless/hands.py

```diff
@@ -0,0 +1,196 @@
+from collections.abc import Callable
+import enum
+import time
+from typing import Any, NamedTuple
+
+import cv2
+from google.protobuf.internal.containers import RepeatedCompositeFieldContainer
+from mediapipe.python.solutions.hands import HandLandmark, Hands
+import numpy as np
+from pydantic import BaseModel
+
+from touchless.gestures.clicks import *
+from touchless.gestures.fingers import *
+from touchless.gestures.hand import *
+from touchless.gestures.pinches import *
+
+from touchless.utils.landmarks import Point, HandLandmarkPoints
+
+
+class HandType(enum.StrEnum):
+    """An enumeration representing types of hands."""
+    RIGHT = "right"
+    LEFT = "left"
+
+
+class HandTrackingData(BaseModel):
+    """A class representing hand tracking data."""
+    is_hand_detected: bool = False
+    hand_confidence: float | None = None
+    keypoints: HandLandmarkPoints | None = None
+    timestamp_ns: int | None = None
+
+
+class GestureTrackingData(BaseModel):
+    """A class representing gesture tracking data."""
+    is_detected: bool
+    gesture_confidence: float
+    timestamp_ns: int
+    datapoints: Any = None
+
+
+class HandGesture(BaseModel):
+    """A class representing a hand gesture."""
+    name: str
+    data: GestureTrackingData
+    provider: str
+
+
+class Hand(BaseModel):
+    """A class representing a hand."""
+    type: HandType
+    data: HandTrackingData = HandTrackingData()
+    required_gestures: list[str] | None = None
+    gestures: list[HandGesture] = []
+
+
+class HandTrackingProvider:
+    """A class for hand tracking."""
+    def __init__(self) -> None:
+        self._hands_processor = Hands()
+
+    def update(self, frame: np.ndarray, right_hand: Hand, left_hand: Hand | None = None) -> None:
```

Comment:
Does it make sense to make `rigth_hand` and `left_hand` a property of the `HandTrackingProvider`?

Reply:
I guese it's greate idea. It needs to think properly about it. Also maybe there it make sense to rename {`HandTrackingProvider` -> `HandProvider`} or implement separate `HandProvider` alongside `HandTrackingProvider` , since it will be about hands, not about hands tracking data only.